js/scenes/Compass.js

```diff
@@ -21,25 +30,40 @@ export default class Debug extends Component {
     };
   }
 
-  componentDidMount() {
-    navigator.geolocation.getCurrentPosition( (position) => {
-      var initialPosition = JSON.stringify(position);
-      this.setState({initialPosition});
+  componentWillReceiveProps({ geolocation }) {
+    if (!geolocation) {
+      return;
     }
-      , (error) => console.log(JSON.stringify(error)),
-      {enableHighAccuracy: true, timeout: 20000, maximumAge: 1000}
-    );
-    this.watchID = navigator.geolocation.watchPosition((position) => {
-      this.setState({
-        latitude: position.coords.latitude,
-        longitude: position.coords.longitude,
-      });
+
+    this.setState({
+      latitude: geolocation.value.coords.latitude,
+      longitude: geolocation.value.coords.longitude,
     });
   }
 
-  // Watch for position for testing the compass
-  componentWillUnmount() {
-    navigator.geolocation.clearWatch(this.watchID);
+  renderLocation({ status }) {
+    switch(status) {
+      case EMPTY: return this.renderGeoMessage('No data');
+      case GEO_ADQUIRED: return this.renderGeo();
+      case GEO_ADQUIRING: return this.renderGeoMessage('Adquiring geolocation ...');
+      case GEO_ERROR: return this.renderGeoMessage('Error getting geolocation');
+    }
+  }
+
+  renderGeoMessage(msg) {
+    return(
+      <Text>{msg}</Text>
+    );
+  }
+
+  renderGeo() {
+    var when = new Date(this.props.geolocation.value.timestamp);
+    when = '' + when;
```

Comment:
Try not to use `var`, we may disable its usage in the future.
Also `var when = ``${new Date(this.props.geolocation.value.timestamp)}``;` seems more elegant to me.